x/slashing/types/genesis.go

```diff
@@ -7,31 +7,40 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// GenesisState - all slashing state that must be provided at genesis
-type GenesisState struct {
-	Params       Params                          `json:"params" yaml:"params"`
-	SigningInfos map[string]ValidatorSigningInfo `json:"signing_infos" yaml:"signing_infos"`
-	MissedBlocks map[string][]MissedBlock        `json:"missed_blocks" yaml:"missed_blocks"`
-}
-
 // NewGenesisState creates a new GenesisState object
 func NewGenesisState(
 	params Params, signingInfos map[string]ValidatorSigningInfo, missedBlocks map[string][]MissedBlock,
 ) GenesisState {
 
+	var si = make([]SigningInfos, len(signingInfos))
+
+	index := 0
+	for address, signingInfo := range signingInfos {
+		si[index] = SigningInfos{
+			Address:      address,
+			SigningInfos: signingInfo,
+		}
+		index++
+	}
+
+	var validatorMissedBlocks = make([]ValidatorMissedBlocks, len(missedBlocks))
+
+	index = 0
+	for address, validatorMissedBlock := range missedBlocks {
```

Comment:
Do we have any performance advantage of using this way, rather than migrating ExportGenesis to use new structs?

Reply:
I only suggested this if we needed O(1) lookup in business logic, if it's only used on `InitGenesis`, then we don't need to do this.